sample-submodule/src/main/java/com/airbnb/android/submodule/Payments.kt

```diff
@@ -101,3 +102,29 @@ val creditCardOffsetMap = object : OffsetMap {
 fun CreditCardComponentPreview() {
     CreditCardComponent("John Doe")
 }
+
+
+data class DemoDataClass(
+    val name: String,
+    val age: Int
+)
+
+class NewParameterProvider : PreviewParameterProvider<DemoDataClass> {
```

Comment:
I see some opportunity to ship a small class that wraps this boilerplate with a basic list implementation (having to provide the count is annoying...). potentially a class that takes the list of examples in the constructor
```
class NewParameterProvider : PreviewParameterList {
 listOf(
            DemoDataClass("John", 12),
            DemoDataClass("Doe", 20)
 )
}
```

i guess we still want it to be lazy, so they aren't instantiated on each function call, so the constructor can take a lambda that returns a list

Reply:
providing the count is actually optional so I can just remove that line. Also Preview would not know how to instantiate this class with data so passing parameters wouldn't be possible I believe @elihart